server/application/application.go

```diff
@@ -303,21 +300,77 @@ func (s *Server) List(ctx context.Context, q *application.ApplicationQuery) (*v1
 			newItems = append(newItems, *a)
 		}
 	}
+	// App counters are based on the filtered list of applications
+	for _, app := range newItems {
+		stats.Total++
+		stats.TotalByHealthStatus[app.Status.Health.Status]++
+		stats.TotalBySyncStatus[app.Status.Sync.Status]++
+		if app.Spec.SyncPolicy != nil && app.Spec.SyncPolicy.Automated != nil {
+			stats.AutoSyncEnabledCount++
+		}
+	}
 
 	// Sort found applications by name
 	sort.Slice(newItems, func(i, j int) bool {
-		return newItems[i].Name < newItems[j].Name
+		return less(newItems[i], newItems[j])
 	})
 
+	if q.Offset != nil && q.Limit != nil {
+		if *q.Offset < 0 || *q.Limit < 0 {
+			return nil, fmt.Errorf("offset %d and limit %d must be a non-negative integer", *q.Offset, *q.Limit)
+		}
+		if *q.Offset >= int64(len(newItems)) || *q.Limit == 0 {
+			newItems = make([]v1alpha1.Application, 0)
+		} else {
+			if *q.Offset+*q.Limit >= int64(len(newItems)) {
+				newItems = newItems[*q.Offset:]
+			} else {
+				newItems = newItems[*q.Offset : *q.Offset+*q.Limit]
+			}
+		}
+	}
+
 	appList := v1alpha1.ApplicationList{
 		ListMeta: metav1.ListMeta{
 			ResourceVersion: s.appInformer.LastSyncResourceVersion(),
 		},
 		Items: newItems,
+		Stats: stats,
 	}
 	return &appList, nil
 }
 
+func (s *Server) getAppsStats(apps []*v1alpha1.Application) v1alpha1.ApplicationListStats {
+	stats := v1alpha1.NewApplicationListStats()
+	destinations := map[v1alpha1.ApplicationDestination]bool{}
+	namespaces := map[string]bool{}
+	labels := map[string]map[string]bool{}
+	for _, app := range apps {
+		if _, ok := destinations[app.Spec.Destination]; !ok {
+			destinations[app.Spec.Destination] = true
+		}
+		if _, ok := namespaces[app.Spec.Destination.Namespace]; !ok {
+			namespaces[app.Spec.Destination.Namespace] = true
+		}
+		for key, value := range app.Labels {
+			if valueMap, ok := labels[key]; !ok {
+				labels[key] = map[string]bool{value: true}
+			} else {
+				valueMap[value] = true
+			}
+		}
+	}
+	stats.Destinations = lo.Keys(destinations)
+	stats.Namespaces = lo.Keys(namespaces)
+	for key, valueMap := range labels {
+		stats.Labels = append(stats.Labels, v1alpha1.ApplicationLabelStats{
+			Key:    key,
+			Values: lo.Keys(valueMap),
+		})
+	}
```

Comment:
If we're only using the `Keys` function, I think I'd rather use a utility function instead of importing an external dependency.

Reply:
Yes indeed, but samber/lo is a Lodash-style go library, I found a lot of code could be replaced with functions provided by the library, should I open a PR to replace all occurence or just let it go?

Reply:
Guessing that the standard lib function https://pkg.go.dev/maps#Keys would cover this need right?

Reply:
Yes, you are right, I removed the util function